base/compiler/optimize.jl

```diff
@@ -48,6 +48,10 @@ intersect!(et::EdgeTracker, range::WorldRange) =
     et.valid_worlds[] = intersect(et.valid_worlds[], range)
 
 push!(et::EdgeTracker, mi::MethodInstance) = push!(et.edges, mi)
+function push!(et::EdgeTracker, @nospecialize(invokesig), mi::MethodInstance)
+    invokesig === nothing && return push!(et.edges, mi)
+    push!(et.edges, invokesig, mi)
+end
```

Comment:
3 arg push feels wrong.
At this point, should we have a new function like `track_edge!` rather than punning on `push!`?

Reply:
```julia
julia> x = Any[]
Any[]

julia> push!(x, "Hi", 2);

julia> x
2-element Vector{Any}:
  "Hi"
 2

julia> versioninfo()
Julia Version 1.8.0-rc1.41
Commit 381eac37ca* (2022-07-06 11:18 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 4 virtual cores
Environment:
  JULIA_CPU_THREADS = 4
```

Reply:
I agree that this feel weird. Having a fallback that makes multiarg push turn into append is different than dispatching on multiple types in a push.